### PR TITLE
Add a ComposeState for _wait_for_sync()

### LINKED ISSUE #2550: Add a ComposeState for waiting on the sync to mirrors
We should add a new ComposeState() to get set when ```_wait_for_sync()``` is called in the composer.

### PR DESCRIPTION
**Pull request by mattiaverga**
Fix #2550 

Signed-off-by: Mattia Verga <mattia.verga@tiscali.it>

### CODE REVIEW

**Review by bowlofeggs — CHANGES_REQUESTED** [review]
Nice! This looks pretty good. The only thing we still need that I can think of is a database migration to extend the enum. There are a couple migrations that do something like this already that you can base your migration off of, for example:

https://github.com/fedora-infra/bodhi/blob/3.10.0/bodhi/server/migrations/versions/2c5d45f0c932_add_a_new_state_for_cleaning_composes.py

You can use alembic to start the template for you like this:

```
$ alembic revision --autogenerate -m "Add a syncing_repo ComposeState enum value."
```

**Comment by mattiaverga** [comment]
I've added the `syncing_repo` value after `punging` instead at the end of the list to maintain the order specified in the docstring (where I added it in the right point of the sequence).
I'm totally new to alembic, but running the upgrade locally returned no errors, so I think it's ok...

**Comment by mattiaverga** [comment]
jenkies test

File: bodhi/server/migrations/versions/68f9ccb1f388_add_a_syncing_repo_composestate_enum_.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Copyright (c) YEAR AUTHOR
```
Comment (bowlofeggs) [inline bodhi/server/migrations/versions/68f9ccb1f388_add_a_syncing_repo_composestate_enum_.py]: Let's fill in the year and author:

```
# Copyright (c) 2018 Mattia Verga
```

**Review by bowlofeggs — CHANGES_REQUESTED** [review]
Looks great - just need you to claim copyright on your migration and we should be good to merge.

File: bodhi/server/migrations/versions/68f9ccb1f388_add_a_syncing_repo_composestate_enum_.py (hunk 1 above)
Comment (bowlofeggs) [reply bodhi/server/migrations/versions/68f9ccb1f388_add_a_syncing_repo_composestate_enum_.py]: This is a minor enough of a change that I'll just force push your branch and fix it for you. Fix incoming!

**Review by bowlofeggs — DISMISSED** [review]

**Comment by bowlofeggs** [comment]
jenkies test

**Review by bowlofeggs — APPROVED** [review]

**Comment by bowlofeggs** [comment]
jenkies test

**Comment by bowlofeggs** [comment]
Mergify seems stuck, so I'm going to manually rebase and merge this one.

**Pull request merged**